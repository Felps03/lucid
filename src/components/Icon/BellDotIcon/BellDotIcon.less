@import (reference) '../../../styles/variables.less';
@import (reference) '../../../styles/mixins.less';

.@{prefix}-BellDotIcon {
	&-background {
		fill: @featured-color-info;
		stroke: none;
	}
}
