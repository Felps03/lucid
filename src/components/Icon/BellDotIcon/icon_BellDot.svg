<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M5.5 12.502v.5c0 1.381 1.119 2.5 2.5 2.5s2.5-1.119 2.5-2.5l.003-.495M3.056 7.432c-.54 2.214-1.556 3.07-1.556 3.07v2h13v-2s-1.884-1.587-1.884-6.024c0-.901-.909-3.976-4.616-3.976-.279 0-.543.017-.791.05" fill="none" stroke="#000" stroke-width="1.3" stroke-miterlimit="10"/><path d="M3.5 6.152c-1.737 0-3.15-1.413-3.15-3.15S1.763-.148 3.5-.148s3.15 1.413 3.15 3.15-1.413 3.15-3.15 3.15z"/><path d="M3.5-.148c1.737 0 3.15 1.413 3.15 3.15s-1.413 3.15-3.15 3.15S.35 4.739.35 3.002 1.763-.148 3.5-.148" fill="#587eba"/></svg>