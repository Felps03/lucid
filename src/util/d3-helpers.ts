import { Selection } from 'd3-selection';
import _ from 'lodash';
import * as d3Array from 'd3-array';
import ReactDOM from 'react-dom';

enum Orientation {
	TOP = 'TOP',
	RIGHT = 'RIGHT',
	BOTTOM = 'BOTTOM',
	LEFT = 'LEFT',
}

export type IXAxisRenderProp = (xValue: string) => JSX.Element;

const setLineAttributes = (
	selection: any,
	orientation: any,
	range: [number, number]
) => {
	const [min, max] = range;

	if (orientation === Orientation.BOTTOM || orientation === Orientation.TOP) {
		selection
			.attr('x1', min)
			.attr('x2', max)
			.attr('y1', 0)
			.attr('y2', 0);
	} else {
		selection
			.attr('x1', 0)
			.attr('x2', 0)
			.attr('y1', min)
			.attr('y2', max);
	}
};

const setTickAttributes = (
	selection: any,
	orientation: Orientation,
	scale: any,
	tickSize: number
) => {
	const x1 = 0;
	let x2 = 0;
	const y1 = 0;
	let y2 = 0;
	switch (orientation) {
		case Orientation.BOTTOM:
			y2 = tickSize || 5;
			break;
		case Orientation.TOP:
			y2 = -tickSize || 5;
			break;
		case Orientation.RIGHT:
			x2 = tickSize || 5;
			break;
		case Orientation.LEFT:
			x2 = -(tickSize || 5);
			break;
	}
	selection
		.attr('x1', x1)
		.attr('x2', x2)
		.attr('y1', y1)
		.attr('y2', y2);
};

const lucidXAxis = (
	selection: Selection<SVGElement, {}, null, undefined>,
	{
		xScale,
		tickSize,
		xAxisRenderProp,
	}: {
		xScale: any;
		tickSize: number;
		xAxisRenderProp?: IXAxisRenderProp;
	}
): void => {
	const range = xScale.range();
	const domain = xScale.domain();
	const rMin = Number(d3Array.min(range)) || 0;
	const rMax = Number(d3Array.max(range)) || 0;
	const axisGroup = selection.append('g').classed('axis', true);
	axisGroup
		.append('line')
		.attr('stroke', 'black')
		.call(setLineAttributes, 'TOP', [rMin, rMax]);
	const ticks = axisGroup
		.selectAll('.ticks')
		.data(domain)
		.order();

	const tick = ticks
		.enter()
		.append('g')
		.classed('ticks', true)
		.attr('transform', (d: any) => `translate( ${xScale(d)}, 0)`);
	tick.append('line').call(setTickAttributes, 'TOP', xScale, tickSize);
	tick
		.append('foreignObject')
		.style('overflow', 'visible')
		.style('display', 'flex')
		.attr('x', 0)
		.attr('y', 0)
		.attr('height', 1)
		.attr('width', 1);
	if (!xAxisRenderProp) {
		tick.append('text').text((d: any) => d);
	} else {
		const tickObj = tick.selectAll('foreignObject').append('xhtml:div');
		tickObj
			.style('width', `${60}px`)
			.attr('id', (d: any) => d)
			.classed('tickRender', true)
			.html((xValue: string | any, num, node) => {
				if (xValue !== '' && !_.isNil(xValue)) {
					ReactDOM.render(
						// @ts-ignore
						xAxisRenderProp(xValue),
						node[0]
					);
				}
				// console.log(d,node[0])
				else return xValue;
			});
	}
};

export { lucidXAxis };
