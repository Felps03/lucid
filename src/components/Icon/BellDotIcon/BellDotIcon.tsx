import _ from 'lodash';
import React from 'react';
import Icon, { IIconProps, propTypes as iconPropTypes } from '../Icon';
import { lucidClassNames } from '../../../util/style-helpers';
import { omitProps } from '../../../util/component-types';

const cx = lucidClassNames.bind('&-BellDotIcon');

interface IBellDotIconProps extends IIconProps {}

export const BellDotIcon = ({
	className,
	...passThroughs
}: IBellDotIconProps) => {
	return (
		<Icon
			{...omitProps(
				passThroughs,
				undefined,
				_.keys(BellDotIcon.propTypes),
				false
			)}
			{..._.pick(passThroughs, _.keys(iconPropTypes))}
			className={cx('&', className)}
		>
			<path d='M5.5 12.502v.5c0 1.381 1.119 2.5 2.5 2.5s2.5-1.119 2.5-2.5l.003-.495M3.056 7.432c-.54 2.214-1.556 3.07-1.556 3.07v2h13v-2s-1.884-1.587-1.884-6.024c0-.901-.909-3.976-4.616-3.976-.279 0-.543.017-.791.05' />
			<path d='M3.5 6.152c-1.737 0-3.15-1.413-3.15-3.15S1.763-.148 3.5-.148s3.15 1.413 3.15 3.15-1.413 3.15-3.15 3.15z' />
			<path
				className={cx('&-background')}
				d='M3.5-.148c1.737 0 3.15 1.413 3.15 3.15s-1.413 3.15-3.15 3.15S.35 4.739.35 3.002 1.763-.148 3.5-.148'
			/>
		</Icon>
	);
};

BellDotIcon.displayName = 'BellDotIcon';
BellDotIcon.peek = {
	description: `
		Typically used for notifications that have an unread message.
	`,
	categories: ['visual design', 'icons'],
	extend: 'Icon',
	madeFrom: ['Icon'],
};
BellDotIcon.propTypes = iconPropTypes;
BellDotIcon.defaultProps = Icon.defaultProps;

export default BellDotIcon;
