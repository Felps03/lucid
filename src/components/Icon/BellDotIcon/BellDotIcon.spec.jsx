import { icons, common } from '../../../util/generic-tests';

import BellDotIcon from './BellDotIcon';

describe('BellDotIcon', () => {
	common(BellDotIcon);
	icons(BellDotIcon);
});
