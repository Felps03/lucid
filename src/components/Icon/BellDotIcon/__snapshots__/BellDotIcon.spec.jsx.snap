// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BellDotIcon [common] example testing should match snapshot(s) for 1.default 1`] = `
<Icon
  aspectRatio="xMidYMid meet"
  className="lucid-BellDotIcon"
  color="primary"
  isClickable={false}
  isDisabled={false}
  onClick={[Function]}
  onSelect={[Function]}
  size={16}
  viewBox="0 0 16 16"
>
  <path
    d="M5.5 12.502v.5c0 1.381 1.119 2.5 2.5 2.5s2.5-1.119 2.5-2.5l.003-.495M3.056 7.432c-.54 2.214-1.556 3.07-1.556 3.07v2h13v-2s-1.884-1.587-1.884-6.024c0-.901-.909-3.976-4.616-3.976-.279 0-.543.017-.791.05"
  />
  <path
    d="M3.5 6.152c-1.737 0-3.15-1.413-3.15-3.15S1.763-.148 3.5-.148s3.15 1.413 3.15 3.15-1.413 3.15-3.15 3.15z"
  />
  <path
    className="lucid-BellDotIcon-background"
    d="M3.5-.148c1.737 0 3.15 1.413 3.15 3.15s-1.413 3.15-3.15 3.15S.35 4.739.35 3.002 1.763-.148 3.5-.148"
  />
</Icon>
`;
